### PR TITLE
[NEW] Add permission `view-broadcast-member-list`

### PR DESCRIPTION
**Pull request by cardoso**
Makes the permission to view members list in broadcast channels configurable instead of hardcoding which roles can do it

Also unauthorizes users without permissions from fetching the members list through REST & Real-Time APIs.

### CODE REVIEW

**Comment by rafaelks** [comment]
Nice one @cardoso!

File: packages/rocketchat-api/server/v1/groups.js [hunk 1]
```diff
@@ -380,6 +380,11 @@ RocketChat.API.v1.addRoute('groups.listAll', { authRequired: true }, {
 RocketChat.API.v1.addRoute('groups.members', { authRequired: true }, {
 	get() {
 		const findResult = findPrivateGroupByIdOrName({ params: this.requestParams(), userId: this.userId });
+
+		if (findResult._room.broadcast && !RocketChat.authz.hasPermission(this.userId, 'view-broadcast-member-list')) {
```
Comment (rafaelks) [inline packages/rocketchat-api/server/v1/groups.js]: Can't we use a "constant" on the name of the permission? Looks like we're repeating it many times.

File: packages/rocketchat-authorization/server/startup.js [hunk 2]
```diff
@@ -67,7 +67,8 @@ Meteor.startup(function() {
 		{ _id: 'view-statistics',               roles : ['admin'] },
 		{ _id: 'view-user-administration',      roles : ['admin'] },
 		{ _id: 'preview-c-room',                roles : ['admin', 'user', 'anonymous'] },
-		{ _id: 'view-outside-room',             roles : ['admin', 'owner', 'moderator', 'user'] }
+		{ _id: 'view-outside-room',             roles : ['admin', 'owner', 'moderator', 'user'] },
+		{ _id: 'view-broadcast-member-list',    roles : ['admin', 'owner', 'moderator'] }
```
Comment (rafaelks) [inline packages/rocketchat-authorization/server/startup.js]: Also repeating many times the roles here and the name of the permissions.

File: packages/rocketchat-lib/client/defaultTabBars.js [hunk 3]
```diff
@@ -33,7 +33,7 @@ RocketChat.TabBar.addButton({
 			return true;
 		}
 
-		return RocketChat.authz.hasRole(Meteor.userId(), ['admin', 'moderator', 'owner'], rid);
+		return RocketChat.authz.hasAllPermission('view-broadcast-member-list', rid);
```
Comment (rafaelks) [inline packages/rocketchat-lib/client/defaultTabBars.js]: 👌

File: packages/rocketchat-api/server/v1/groups.js (hunk 1 above)
Comment (cardoso) [reply packages/rocketchat-api/server/v1/groups.js]: We'd have to do that for all other permissions too.

But there wouldn't be much benefit with this being Javascript anyway. ¯\_(ツ)_/¯

**Comment by ggazzo** [comment]
dont care about constant/string

**Comment by cardoso** [comment]
I made the changes you requested @ggazzo 

Can you check if they're ok?

**Comment by ggazzo** [comment]
@cardoso done ;)

**Pull request merged**